tests/test_update_ssh_keys.rs

```diff
@@ -0,0 +1,524 @@
+    iYAcEJ/YbKEnjtJzaAmQNpyh3rCWuOcGPTevjAZIkl+zEc+/N7tCW1e\
+    uDYm6IXZ8LEQyTUQUdU4pZ2OgAAAIABk1ZA3+TiCMaoAafNVUZ7zwqk\
+    888yVOgsJ7HGGDGRMo5ytr2SUJB7QWsLX6Un/Zbu32nXsAqtqagxd6F\
+    Ies98TSekMh/hAv9uK92mEsXSINXOeIMKRedqOyPgk5IEOsFpxAUO4T\
+    xpYToeuM8HRemecxw2eIFHnax+mQqCsi7FgQ== core@valid2",
+                ),
+                (
+                    "valid3",
+                    "\
+    sk-ssh-ed25519@openssh.com AAAAGnNrLXNzaC1lZDI1NTE5QG9w\
+    ZW5zc2guY29tAAAAIEX/dQ0v4127bEo8eeG1EV0ApO2lWbSnN6RWusn\
+    /NjqIAAAABHNzaDo= demos@siril",
+                ),
+                (
+                    "bad",
+                    "ssh-bad this-not-a-key core@bad",
+                ),
+            ].iter().cloned().collect();
+
+        static ref FINGERPRINTS: HashMap<&'static str, &'static str> =
+            [
+                (
+                    "valid1",
+                    "SHA256:yZ+o48h6quk9c+JVgJ/Zq4S5u4LUk6TSpneHKkmM9KY",
+                ),
+                (
+                    "valid2",
+                    "SHA256:RP5k1AybZ1kollIAnpUavr1v1nfZ0yloKvI46AMDPkM",
+                ),
+                (
+                    "valid3",
+                    "SHA256:U8IKRkIHed6vFMTflwweA3HhIf2DWgZ8EFTm9fgwOUk",
+                ),
+            ].iter().cloned().collect();
+    }
+
+    // TestContext holds path to a temporary directory used by the current test.
+    struct TestContext {
+        ssh_dir: PathBuf,
+    }
+
+    // Automatically clean up ssh_dir when each test finished.
+    impl Drop for TestContext {
+        fn drop(&mut self) {
+            _ = fs::remove_dir_all(PathBuf::from(self.ssh_dir.clone()));
+        }
+    }
+
+    fn assert_has_keys(ssh_dir: &str, keys: Vec<&str>, assert_no_key: bool) {
+        let authkeyspath: PathBuf = PathBuf::from(format!("{}/authorized_keys", ssh_dir));
+        let authkeysfile = File::open(authkeyspath.clone());
+        assert!(authkeysfile.is_ok());
+        let mut authkeystext = String::new();
+        authkeysfile.unwrap().read_to_string(&mut authkeystext).expect(format!("unable to read a file {}.", authkeyspath.to_str().unwrap_or_default()).as_str());
+        assert!(authkeystext.starts_with("# auto-generated"));
+
+        for key in &keys {
+            if assert_no_key {
+                assert!(authkeystext.contains(&*TEST_KEYS[key]));
+            } else {
+                assert!(!authkeystext.contains(&*TEST_KEYS[key]));
+            }
+        }
+
+        // NOTE: the tests below causes to all the tests to fail, as it asserts
+        // that all other non-contained keys not included in authorized_keys.
+        // Apparently it fails because authorized_keys still has the other keys.
+
+        //        for (key, _) in &*TEST_KEYS {
+        //            if keys.contains(&key) {
+        //                continue;
+        //            }
+        //            assert!(!authkeystext.contains(&*TEST_KEYS[key]));
+        //        }
```

Comment:
Is this still a problem and if yes, would it help to guard it with assert_no_key?

Reply:
Yes, it is still a problem, does not run as expected, guarding with the flag does not help.

Reply:
It fails because `test_add_bad` passes the wrong parameter.

Reply:
Thanks, done.